lib/libc/Kconfig

```diff
@@ -62,7 +62,7 @@ menu "C Library"
 choice LIBC_IMPLEMENTATION
 	prompt "C Library Implementation"
 	default EXTERNAL_LIBC if NATIVE_BUILD && !(NATIVE_LIBRARY && POSIX_API)
-	default PICOLIBC
+	default PICOLIBC if !(REQUIRES_FULL_LIBC)
```

Comment:
What are you doing, no

Reply:
@thedjnK ,  the code original list as below
```
choice LIBC_IMPLEMENTATION
        prompt "C Library Implementation"
        default EXTERNAL_LIBC if NATIVE_BUILD && !(NATIVE_LIBRARY && POSIX_API)
        default PICOLIBC
        default NEWLIB_LIBC if REQUIRES_FULL_LIBC
        default MINIMAL_LIBC
```
and even we define the REQUIRES_FULL_LIBC, LIBC_IMPLEMENTATION is still PICOLIBC, so this is a strange here, either we change the 
```
 default NEWLIB_LIBC if REQUIRES_FULL_LIBC
```
to 
```
 default NEWLIB_LIBC
```
or do what I change above. 

Besides, I do think force back the newlibc to solve this problem is not good. and I will try keith-packard's recommandations.

Reply:
It is this way on purpose and it is correct, it will use external if needed, it will use picolibc if it can (which should be the default if at all possible), it will then fall back to newlibc if picolibc cannot be used or minimal libc if a full libc is not needed, leave the ordering alone

Reply:
ok